### PR TITLE
When adding results from query operations, discard results when one or more partitions have already been processed

### PR DESCRIPTION
**Pull request by vbekiaris**
When aggregating results, do not take into account results for partition IDs which have already been accounted for. This will cause separate per-partition `QueryPartitionOperation`s to be executed

Fixes over-reporting values in #6471

### CODE REVIEW

**Comment by vbekiaris** [comment]
run-lab-run

File: hazelcast/src/main/java/com/hazelcast/map/impl/query/MapQueryEngineImpl.java [hunk 1]
```diff
@@ -526,6 +526,11 @@ protected void addResultsOfPagingPredicate(List<Future<QueryResult>> futures, Co
 
             Collection<Integer> tmpPartitionIds = queryResult.getPartitionIds();
```
Comment (tombujok) [inline hazelcast/src/main/java/com/hazelcast/map/impl/query/MapQueryEngineImpl.java]: Maybe queryPartitions? tmpPartitions is kind of generic

File: hazelcast/src/test/java/com/hazelcast/map/impl/query/QueryIndexMigrationTest.java [hunk 2]
```diff
@@ -160,7 +160,7 @@ public void testQueryWithIndexesWhileMigrating() throws Exception {
     /**
      * test for issue #359
      */
-    @Test(timeout = MINUTE)
+    @Test
```
Comment (tombujok) [inline hazelcast/src/test/java/com/hazelcast/map/impl/query/QueryIndexMigrationTest.java]: That may be risky, maybe let's set 2 minutes?

File: hazelcast/src/main/java/com/hazelcast/map/impl/query/MapQueryEngineImpl.java (hunk 1 above)
Comment (vbekiaris) [inline hazelcast/src/main/java/com/hazelcast/map/impl/query/MapQueryEngineImpl.java]: Yes, that's the variable's name in previous code but it also struck me as not-fitting so will fix it.

File: hazelcast/src/test/java/com/hazelcast/map/impl/query/QueryIndexMigrationTest.java (hunk 2 above)
Comment (vbekiaris) [inline hazelcast/src/test/java/com/hazelcast/map/impl/query/QueryIndexMigrationTest.java]: I thought 2 minutes is the default timeout in Hazelcast test runners, or is this not the case?

Comment (vbekiaris) [inline hazelcast/src/test/java/com/hazelcast/map/impl/query/QueryIndexMigrationTest.java]: After all, this is not the case, added 2 minutes timeout to test definition. Thanks for pointing that out!

**Comment by tombujok** [comment]
:+1: looks good!

**Comment by ahmetmircik** [comment]
👍

**Pull request merged**